system/lib/wasmfs/wasmfs.cpp

```diff
@@ -192,6 +223,10 @@ __wasi_fd_t __syscall_open(long pathname, long flags, long mode) {
     canWrite = true;
   }
 
+  // TODO: remove assert when all functionality is complete.
+  // Currently implement O_APPEND, O_TRUNC
+  assert(!(flags & O_DSYNC) && !(flags & O_NOCTTY) && !(flags & O_NONBLOCK));
```

Comment:
Instead of asserting that all unsupported flags are not used (which requires enumerating all the unsupported flags), I would go the other way and assert that only supported flags are used. It should be easier to reliably enumerate supported flags than unsupported flags. Something like `assert((flags & ~(all | supported | flags)) == 0);` should work.